docs/sphinx/sections/api/apidocs/metadata.rst

```diff
@@ -0,0 +1,65 @@
+.. currentmodule:: dagster
+
+Metadata
+========
+
+Dagster uses metadata to communicate arbitrary user-specified metadata about structured
+events.
+
+.. autoclass:: MetadataValue
+
+.. autoclass:: MetadataEntry
+
+Metadata types
+^^^^^^^^^^^^^^
+
+All metadata types inherit from `MetadataValue`. The following types are defined:
+
+.. autoclass:: DagsterAssetMetadataValue
+
+.. autoclass:: DagsterRunMetadataValue
+
+.. autoclass:: FloatMetadataValue
+
+.. autoclass:: IntMetadataValue
+
+.. autoclass:: JsonMetadataValue
+
+.. autoclass:: MarkdownMetadataValue
+
+.. autoclass:: PathMetadataValue
+
+.. autoclass:: NotebookMetadataValue
+
+.. autoclass:: PythonArtifactMetadataValue
+
+.. autoclass:: TableColumnLineageMetadataValue
+
+.. autoclass:: TableMetadataValue
+
+.. autoclass:: TableSchemaMetadataValue
+
+.. autoclass:: TextMetadataValue
+
+.. autoclass:: TimestampMetadataValue
+
+.. autoclass:: UrlMetadataValue
+
+Tables
+^^^^^^
+
+These APIs provide the ability to express column schemas (`TableSchema`), rows/records (`TableRecord`), and column lineage (`TableColumnLineage`) in Dagster.
```

Comment:
- Should we be specific and say `... and column lineage in Dagster as metadata`?
- Do we need to specify anything about ops and/or assets? These were originally on the ops page, so not completely sure.

Reply:
I think the UI portion should be mentioned in the concepts page, rather than the API docs.